easybuild/tools/github.py

```diff
@@ -595,8 +595,38 @@ def _easyconfigs_pr_common(paths, start_branch=None, pr_branch=None, target_acco
 
     _log.debug("git status: %s", git_repo.git.status())
 
-    # copy files to right place
-    file_info = copy_easyconfigs(paths, os.path.join(git_working_dir, pr_target_repo))
+    # seperate easyconfigs and patch files
+    ec_paths = [p for p in paths if p.endswith('.eb') or not is_patch_file(p)]
+    patch_paths = [p for p in paths if p not in ec_paths]
+
+    # copy easyconfig files to right place
+    target_dir = os.path.join(git_working_dir, pr_target_repo)
+    print_msg("copying easyconfigs to %s..." % target_dir)
+    file_info = copy_easyconfigs(ec_paths, target_dir)
+
+    # figure out to which software name patches relate, and copy them to the right place
+    if patch_paths:
+        patch_specs = []
+        print_msg("determining software names for patch files...")
+        for patch_path in patch_paths:
+            soft_name = None
+            patch_file = os.path.basename(patch_path)
+
+            # consider patch lists of easyconfigs being provided
+            for ec in file_info['ecs']:
+                if patch_file in ec['patches']:
+                    soft_name = ec.name
+                    break
+
+            # FIXME: try harder if software name wasn't found yet?
```

Comment:
@Caylo hmm, ideally, this should be tackled too...

As it is, only the easyconfig files that are being tweaked are being considered when trying to figure out which software the patch belongs too, but this may not be enough in practice

For example:

```
eb --new-pr foo.eb
# oops, forgot patch
eb --update-pr 1234 foo.patch
```

So, I think we should fall back to scanning all easyconfigs, and check which patches they require... This may take a while, but it's the best we can do imho (we can't rely on the patch name).